packages/features/bookings/components/event-meta/Details.tsx

```diff
@@ -0,0 +1,167 @@
+import { Fragment } from "react";
+import React from "react";
+
+import { getEventLocationType } from "@calcom/app-store/locations";
+import classNames from "@calcom/lib/classNames";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import {
+  FiInfo,
+  FiClock,
+  FiMapPin,
+  FiCheckSquare,
+  FiRefreshCcw,
+  FiCreditCard,
+} from "@calcom/ui/components/icon";
+
+import { PublicEvent, EventDetailBlocks } from "../../types";
+import { EventDuration } from "./Duration";
+import { EventOccurences } from "./Occurences";
+import { EventPrice } from "./Price";
+
+type EventDetailsPropsBase = {
+  event: PublicEvent;
+  className?: string;
+};
+
+type EventDetailDefaultBlock = {
+  blocks?: EventDetailBlocks[];
+};
+
+// Rendering a custom block requires passing a name prop,
+// which is used as a key for the block.
+type EventDetailCustomBlock = {
+  blocks?: React.FC[];
+  name: string;
+};
+
+type EventDetailsProps = EventDetailsPropsBase & (EventDetailDefaultBlock | EventDetailCustomBlock);
+
+interface EventMetaProps {
+  icon: React.FC<{ className: string }> | string;
+  children: React.ReactNode;
+  // Emphasises the text in the block. For now only
+  // applying in dark mode.
+  highlight?: boolean;
+  contentClassName?: string;
+  className?: string;
+}
+
+/**
+ * Default order in which the event details will be rendered.
+ */
+const defaultEventDetailsBlocks = [
+  EventDetailBlocks.DESCRIPTION,
+  EventDetailBlocks.REQUIRES_CONFIRMATION,
+  EventDetailBlocks.DURATION,
+  EventDetailBlocks.OCCURENCES,
+  EventDetailBlocks.LOCATION,
+  EventDetailBlocks.PRICE,
+];
+
+export const EventMetaBlock = ({
```

Comment:
Event meta block is reusable component that can be used to show event information — it takes in an icon and children, mainly ensuring the layout is correct. It's also used in the EventDetails component down below.